portality/migrate/subjects/remove_duplicate_subjects.py

```diff
@@ -0,0 +1,85 @@
+""" Scroll through articles, removing duplicate subjects. No evidence of same in journals """
+
+import esprit
+from portality import models
+from datetime import datetime
+from portality.core import app
+from portality.util import unicode_dict
+
+batch_size = 1000
+
+
+def rem_dup_sub(write_changes=False):
+
+    # Connection to the ES index
+    conn = esprit.raw.make_connection(None, 'localhost', 9200, 'doaj')
+
+    write_batch = []
+    updated_count = 0
+    same_count = 0
+
+    # Scroll though all articles in the index
+    for a in esprit.tasks.scroll(conn, 'article'):
+        try:
+            article_model = models.Article(_source=a)
+
+            # Remove the duplicates
+            subjects = unicode_dict(article_model.bibjson().subjects())
+            newsub = []
+
+            for sub in subjects:
+                if sub not in newsub:
+                    newsub.append(sub)
```

Comment:
I can't think of why what you've done wouldn't work, I just know of list(set(orig_list)) as the goto for list deduplication in Python